framework/src/media/audio/audio_manager.c

```diff
@@ -146,44 +152,129 @@ static const struct audio_samprate_map_entry_s g_audio_samprate_entry[] = {
 /****************************************************************************
  * Private Function Prototypes
  ****************************************************************************/
+static audio_manager_result_t get_audio_card_in(audio_card_operation_t card_operation, audio_handler_t **handler);
+static audio_manager_result_t get_audio_card_out(audio_card_operation_t card_operation, audio_handler_t **handler);
 static audio_manager_result_t find_audio_card(audio_card_type_t card_type);
-static int get_actual_audio_in_card_id(void);
-static int get_actual_audio_out_card_id(void);
-static void get_hardware_params(audio_card_info_t *card, audio_card_type_t card_type);
+static void get_hardware_params(audio_handler_t *handler, audio_card_type_t card_type);
 static audio_manager_result_t get_supported_sample_rate(int fd, uint8_t *sample_type, audio_card_type_t type);
-static uint32_t get_closest_samprate(unsigned origin_samprate, audio_card_type_t type);
-static unsigned int resample_stream_in(audio_card_info_t *cur_card, void *data, unsigned int frames);
-static unsigned int resample_stream_out(audio_card_info_t *cur_card, void *data, unsigned int frames);
+static uint32_t get_closest_samprate(audio_handler_t *handler, unsigned origin_samprate);
```

Comment:
unsigned origin_samprate ??
you missing data type.

Reply:
Right. Modified.